app/src/lib/ci-checks/ci-checks.ts

```diff
@@ -183,37 +183,20 @@ function getCheckRunShortDescription(
 
   const preposition = conclusion === APICheckConclusion.Success ? 'in' : 'after'
 
-  if (durationSeconds !== undefined && durationSeconds > 0) {
-    const duration =
-      durationSeconds < 60
-        ? `${durationSeconds}s`
-        : `${Math.round(durationSeconds / 60)}m`
-    return `${adjective} ${preposition} ${duration}`
+  if (durationMs !== undefined && durationMs > 0) {
+    return `${adjective} ${preposition} ${formatPreciseDuration(durationMs)}`
   }
 
   return adjective
 }
 
 /**
- * Attempts to get the duration of a check run in seconds.
- * If it fails, it returns 0
+ * Attempts to get the duration of a check run in milliseconds. Returns NaN if
+ * parsing either completed_at or started_at fails
  */
-export function getCheckDurationInSeconds(
+export const getCheckDurationInMilliseconds = (
   checkRun: IAPIRefCheckRun | IAPIWorkflowJobStep
-): number {
```

Comment:
Date objects don't throw for invalid date strings, they return a NaN date object so we can simplify this method quite a bit. That coupled with the fact that `getFormattedCheckRunDuration` now handles NaN makes this short and sweet.